File: c/chapitre_2/exo_25.c
#include <stdio.h>
#include <stdlib.h>
#include <sys/wait.h>
#include <time.h>
#include <unistd.h>

int status ;

void filsAttendre()
{
    srand(time(NULL)) ;
    int nbsec = (rand() % 10 ) + 1 ; // Attend entre 1 à 10
    sleep(nbsec); 

    exit(nbsec) ;
}

int main()
{
    __pid_t pidFils ;
    int total = 0 ;

    switch ( pidFils = fork() )
    {
    case -1 :
        perror("Erreur Fork\n");
        exit(-1);
        break;
    case 0 :
        filsAttendre() ;
        break;
    default :
            wait(&status) ;
            total += WEXITSTATUS(status) ;
        break;
    }

    if (pidFils != 0 ) // on est dans le Père
    {
        printf("Pére : durée total : %d sécondes \n", total );
    }

    return 0 ;
}

File: python/chapitre_2/exo25.py
import os
import sys
import random
import time

def filsAttendre():
    random.seed(time.time())
    nbsec = random.randint(1,10)
    time.sleep(nbsec)
    os._exit(nbsec)

def main():
    total = 0
    pidFils = os.fork()

    if pidFils == -1:
        print("Erreur Fork")
        sys.exit(-1)
    elif pidFils == 0:
        filsAttendre()
    else:
        pid, status = os.wait()
        total += os.WEXITSTATUS(status)

    if pidFils != 0:
        print(f"Pére : durée total : {total} sécondes")

if __name__ == "__main__":
    main()